flutter/lib/src/binding_wrapper.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/widgets.dart';
+
+// The methods and properties are modelled after the the real binding class.
+class BindingWrapper {
+  /// Flutter >= 2.12 throws if WidgetsBinding.instance isn't initialized.
+  // When this method is called, it is guaranteed that a binding was already
+  // initialized.
+  WidgetsBinding get instance => _ambiguate(WidgetsBinding.instance);
+
+  /// Initializes the Binding.
+  void ensureInitialized() => WidgetsFlutterBinding.ensureInitialized();
+}
+
+WidgetsBinding _ambiguate(WidgetsBinding? binding) => binding!;
```

Comment:
The problem here is that `BindingBase.checkInstance` can throw a `FlutterError`.
The utils had a try/catch and this one does not.

Reply:
I don't know if this would ever happen tho, since the SDK calls `ensureInitialized` automatically, but better safe than sorry.

Reply:
`WidgetsBinding.instance` throwing indicates that we're not using it correctly. That's also why I introduced both, `ensureInitialized()` and `instance`. Having both makes it obvious when it fails why it failed. (Which is not initializing the binding before using it)

After the first call to `ensureInitialized()`, it's always safe to call `instance`.

I can reintroduce those safeguards, but I don't think it's necessary. The current Flutter version also has the same API like this PR does. If the try-catch is added back, the try catch should not swallow the exception anymore, because that makes it basically impossible to notice or to fix, if it happens to go wrong.

Reply:
Yeah I totally get it, we can always log out in case this happens, but the SDK should never be a reason to degrade people's apps, if we call it automatically and we cause the prob, we should log out and swallow the exception rather than making the App or the SDK unusable.
https://develop.sentry.dev/sdk/philosophy/#degrade-gracefully
So I rather prefer a functional SDK + events without the contexts that depend on it rather than not working at all.
In general, this would never happen, so it's not a biggie, but the wrapper should still allow returning `null`.
Does that make sense?